### PR TITLE
added fluid/qml to qml importpaths in examples/perproject/minimalqmake

### PR DESCRIPTION
**Pull request by rmallah**
(1) Fixes the runtime error  module "Fluid.*" is not installed by
      setting environment variable QML2_IMPORT_PATH from within the
      application.

### CODE REVIEW

**Comment by rmallah** [comment]
Dear Pier , @plfiorini  
could you pls comment on this pull request. 

kind regds
mallah.

File: src/imports/controls/NavigationDrawer.qml [hunk 1]
```diff
@@ -108,6 +108,13 @@ Drawer {
     */
```
Comment (plfiorini) [inline src/imports/controls/NavigationDrawer.qml]: this change should not be part of the patch

File: examples/perproject/minimalqmake/src/main.cpp [hunk 2]
```diff
@@ -6,6 +6,8 @@ int main(int argc, char *argv[])
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
     QGuiApplication app(argc, argv);
 
+    qputenv("QML2_IMPORT_PATH", "../fluid/qml");
```
Comment (plfiorini) [inline examples/perproject/minimalqmake/src/main.cpp]: `engine.addImportPath(QLatin1String("../fluid/qml"))` would be better.

Comment (rmallah) [reply examples/perproject/minimalqmake/src/main.cpp]: i had initially done that only but observed that , fluidicons provider does not work.
QML Image: Failed to get image from provider: image://fluidicons/navigation/close

on digging further it was found that in 

`fluid/src/imports/core/iconsimageprovider.cpp`

```

   const QString targetPath = QStringLiteral("Fluid/Controls/icons");
    const QStringList importPaths = QQmlEngine().importPathList();

    for (const QString &importPath: importPaths) {
        QDir dir(importPath);
        if (dir.exists(targetPath)) {
            QDir targetDir(dir.absoluteFilePath(targetPath));
            QSvgRenderer renderer(targetDir.absoluteFilePath(id + QLatin1String(".svg")));
            QImage image(size, QImage::Format_ARGB32);
            image.fill(Qt::transparent);
            QPainter painter(&image);
            renderer.render(&painter);
            return image;
        }
    }
```
QQmlEngine().importPathList(); does not contain '../fluid/qml' 
so i set the env variable. This patch was actually meant as a workaround
really not sure of the proper solution.

File: src/imports/controls/NavigationDrawer.qml (hunk 1 above)
Comment (rmallah) [reply src/imports/controls/NavigationDrawer.qml]: this file is actually a part of a different commit , nothing to do 
with the issue#103

Comment (plfiorini) [reply src/imports/controls/NavigationDrawer.qml]: @rmallah can you remove the commit from your pull request?

Comment (rmallah) [reply src/imports/controls/NavigationDrawer.qml]: Yeah I will , i now realize that i should create a local branch for every pull request.
I do not know of a easy way to remove a single commit to a pull request . I will delete this commit
and push it to my branch , hopefully git will do the rest. thanks.

File: examples/perproject/minimalqmake/src/main.cpp (hunk 2 above)
Comment (plfiorini) [reply examples/perproject/minimalqmake/src/main.cpp]: That's because a new `QQmlEngine` is created there with the default import path list.
I'm gonna try to see if I can access the actual engine for the icon image provider.

File: src/imports/controls/NavigationDrawer.qml (hunk 1 above)
Comment (plfiorini) [reply src/imports/controls/NavigationDrawer.qml]:
```sh
git checkout lirios_fluid # enter the branch you are submitting as a pull request
# now isolate the nav drawer patch:
git checkout -b feature/navdrawer
git rebase -i HEAD~2
# now from text editor that has automatically opened, delete the first line, save and close
# return to the lirios_fluid branch and update the pull request
git checkout lirios_fluid
git rebase -i HEAD~2
# now from the text editor remove the second line, save and close
git push -f
```

File: examples/perproject/minimalqmake/src/main.cpp (hunk 2 above)
Comment (plfiorini) [reply examples/perproject/minimalqmake/src/main.cpp]: I can't find a way to do it. Can't use `qmlEngine(object)` because `QQmlImageProviderBase` is not a `QObject`.

So for now you should set the `QML2_IMPORT_PATH` environment variable.

Anyway with Qt 5.10, QtQuick Controls 2 has support for icon themes and I plan to take advantage of it so at some point before Fluid 1.0 the image provider might go away.

Comment (plfiorini) [reply examples/perproject/minimalqmake/src/main.cpp]: Err, I'm wrong setting the environment variable will only make the program run from QtCreator but it won't work when the application is deployed.

Adding to the import path instead would work (if it wasn't for the icons image provider issue).

**Comment by rmallah** [comment]
Hi Pier , since this example is mentioned in the README many first timer end up attempting
to use it. In case they are not able to run it  , it makes the learning curve stiff. 

Can we have some interim solution ?

thanks & regards

**Comment by plfiorini** [comment]
Set `QML2_IMPORT_PATH` in the project settings.

**Comment by rmallah** [comment]
thanks will try.

**Comment by rmallah** [comment]
I used QtCreator -> Projects -> Run Environment . But it did not do the trick.

**Comment by plfiorini** [comment]
@rmallah f070eb79ac62e983b7f1e6d44d4c3d52e2ac5f7f should fix it

**Comment by rmallah** [comment]
Hi Pier , 
sounds great! , it is very nice development. 
 i  shall test the change in my environment if everything fine i 
shall close this pull request.
